.github/workflows/build_container_image.yaml

```
@@ -20,20 +20,16 @@ jobs:
         run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
       - name: Set branch name
         id: branch
-        run: echo "::set-output name=name::$(BRANCH_NAME=${GITHUB_REF##*/}; echo ${BRANCH_NAME/release_/}-auto)"
-      - run: docker build . -t galaxy/galaxy-min:${{ steps.vars.outputs.sha_short }} -f .k8s_ci.Dockerfile
-      - run: docker tag galaxy/galaxy-min:${{ steps.vars.outputs.sha_short }} quay.io/galaxy-k8s/galaxy:${{ steps.branch.outputs.name }} && docker tag galaxy/galaxy-min:${{ steps.vars.outputs.sha_short }} quay.io/galaxy-k8s/galaxy:${{ steps.vars.outputs.sha_short }}
+        run: echo "::set-output name=name::$(BRANCH_NAME=${GITHUB_REF##*/}; echo ${BRANCH_NAME/release_/})"
+      - run: docker build . --build-arg GIT_COMMIT=$(git rev-parse HEAD) --build-arg BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ') --build-arg IMAGE_TAG=${{ steps.branch.outputs.name }} --build-arg IMAGE_TAG=${{ steps.branch.outputs.name }} -t galaxy/galaxy-min:${{ steps.branch.outputs.name }} -f .k8s_ci.Dockerfile
+      - run: docker tag galaxy/galaxy-min:${{ steps.branch.outputs.name }} quay.io/galaxy-min/galaxy:${{ steps.branch.outputs.name }}
```

```suggestion
      - name: Set outputs
        id: commit
        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
      - run: docker build . --build-arg GIT_COMMIT=$(git rev-parse HEAD) --build-arg BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ') --build-arg IMAGE_TAG=${{ steps.branch.outputs.name }} --build-arg IMAGE_TAG=${{ steps.branch.outputs.name }} -t galaxy/galaxy-min:${{ steps.branch.outputs.name }} -t quay.io/galaxy-min/galaxy:${{ steps.branch.outputs.name }} -t quay.io/galaxy-k8s/galaxy:${{ steps.commit.outputs.sha_short }} -f .k8s_ci.Dockerfile
      - name: Login to quay.io
        uses: actions-hub/docker/login@master
        env:
          DOCKER_USERNAME: ${{ secrets.K8S_QUAY_USERNAME }}
          DOCKER_PASSWORD: ${{ secrets.K8S_QUAY_PASSWORD }}
          DOCKER_REGISTRY_URL: quay.io
      - name: Push to quay.io with branch name
        uses: actions-hub/docker@master
        with:
          args: push quay.io/galaxy-k8s/galaxy:${{ steps.commit.outputs.sha_short }}
```

I can slack both of you credentials for `K8S_QUAY_USERNAME` and `K8S_QUAY_PASSWORD`

---

That looks good to me, but we're back to tagging images with the commit. Which is also fine with me, but I think you were concerned about this @nuwang? And just to confirm, we want to push to `quay.io/galaxy-k8s/` and `quay.io/galaxy-min/` and they have different credentials (`K8S_QUAY_*` and `QUAY_*`), right ?

---

This is mainly for the gitops preview system right? If you're planning to continue work on that, that'd be great and I'm fine with the idea of pushing to a completely separate, automation managed repo (galaxy-k8s) with commit hashes, and leaving the other repo (galaxy-min) human readable. It would probably be good to clean up older commit hashes if possible in the former case, just to reduce noise and burden on quay.io.

---

@almahmoud Any objections if I call the automated one: galaxy-k8s-auto?

---

No objections to the name change. I edited the suggestion, and I'll make the new org (quay doesn't support name changing) and send new credentials on slack

---

Regarding reducing noise and burden on quay, I made some changes to the suggestion and separated the build (inheriting image though, so it takes less than a second) for the auto one and added an expiration label to it. I pushed a test one to make sure it works, and it seems to properly have taken the expiration date: https://quay.io/repository/galaxy-k8s-auto/galaxy?tab=tags . I chose 3 months by default, mostly arbitrary choice, but seems like more than enough time to finish a PR without having to rebase.